File: ProiectCsharp/MPP/MPP/repository/RepoDBOrganiser.cs
using System.Data;
using MPP.model;
using log4net;
namespace MPP.repository;

public class RepoDBOrganiser : Repository<Organiser, int>
{
    
    private static readonly ILog log = LogManager.GetLogger("SortingTaskDbRepository");

    IDictionary<String, string> props;
    
    public RepoDBOrganiser(IDictionary<String, string> props)
    {
        log.Info("Creating RepoOrganiserDB");
        this.props = props;
    }

    public void add(Organiser entity)
    {
        log.InfoFormat("saving organiser {0}", entity);
        var con = DBUtils.getConnection(props);
        using (var comm = con.CreateCommand())
        {
            comm.CommandText = "insert into Organiser  values (@Id, @Name)";
            var paramId = comm.CreateParameter();
            paramId.ParameterName = "@Id";
            paramId.Value = entity.Id;
            comm.Parameters.Add(paramId);

            var paramName = comm.CreateParameter();
            paramName.ParameterName = "@Name";
            paramName.Value = entity.Name;
            comm.Parameters.Add(paramName);
            
            var result = comm.ExecuteNonQuery();
            log.InfoFormat("Saved {0} instance", result);
            if (result == 0)
            {
                log.Error("No organiser added !");
                throw new Exception("No organiser added !");
            }
        }

    }

    public void delete(Organiser entity)
    {
        log.InfoFormat("deleting organiser {0}", entity);
        IDbConnection con = DBUtils.getConnection(props);
        using (var comm = con.CreateCommand())
        {
            comm.CommandText = "delete from Organiser where Id=@Id";
            
            IDbDataParameter paramId = comm.CreateParameter();
            paramId.ParameterName = "@Id";
            paramId.Value = entity.Id;
            comm.Parameters.Add(paramId);
            
            var dataR = comm.ExecuteNonQuery();
            log.InfoFormat("Deleted {0} instance", dataR);
            if (dataR == 0)
            {
                log.Error("No organiser deleted!");
                throw new Exception("No organiser deleted!");
            }
        }
    }

    public void update(Organiser entity, int id)
    {
        log.InfoFormat("updating organiser {0} with {1}", id,entity);
        IDbConnection con = DBUtils.getConnection(props);
        using (var comm = con.CreateCommand())
        {
            comm.CommandText = "update Organiser set Name=@Name  where Id=@Id";
            
            var paramId = comm.CreateParameter();
            paramId.ParameterName = "@Id";
            paramId.Value = id;
            comm.Parameters.Add(paramId);

            var paramName = comm.CreateParameter();
            paramName.ParameterName = "@Name";
            paramName.Value = entity.Name;
            comm.Parameters.Add(paramName);
            
            var dataR = comm.ExecuteNonQuery();
            log.InfoFormat("Updated {0} instance", dataR);
            if (dataR == 0)
            {
                log.Error("No organiser  updated!");
                throw new Exception("No organiser updated!");
            }
        }
    }

    public Organiser findbyId(int id)
    {
        log.InfoFormat("finding one organiser {0}", id);
        var con = DBUtils.getConnection(props);
        Organiser organiser = new Organiser();
        using (var comm = con.CreateCommand())
        {
            comm.CommandText = "select Id,Name from Organiser where Id=@Id";
            var paramId = comm.CreateParameter();
            paramId.ParameterName = "@Id";
            paramId.Value = id;
            comm.Parameters.Add(paramId);
            using (var dataR = comm.ExecuteReader())
            {
                while (dataR.Read())
                {
                    int idb = dataR.GetInt32(0);
                    string name = dataR.GetString(1);
                    organiser.Id = idb;
                    organiser.Name = name;
                }
            }
        }
        log.InfoFormat("Found {0} instance", organiser);
        return organiser;
    }

    public ICollection<Organiser> findAll()
    {
        log.InfoFormat("finding all organisers");
        IDbConnection con = DBUtils.getConnection(props);
        IList<Organiser> organisers = new List<Organiser>();
        using (var comm = con.CreateCommand())
        {
            comm.CommandText = "select Id,Name from Organiser";
            using (var dataR = comm.ExecuteReader())
            {
                while (dataR.Read())
                {
                    int id = dataR.GetInt32(0);
                    String name = dataR.GetString(1);
                    Organiser organiser = new Organiser(id, name);
                    organisers.Add(organiser);
                }
            }
        }
        log.InfoFormat("found {0} organisers",organisers.Count);
        return organisers;
    }
}

File: JavaSockets/BaskRepos/src/main/java/bask/repos/RepoDBOrganiser.java
package bask.repos;

import bask.model.JdbcUtils;
import bask.model.Organiser;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.Properties;

public class RepoDBOrganiser implements IRepoOrganiser {

    private JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger();

    public RepoDBOrganiser(Properties props) {
        dbUtils = new JdbcUtils(props);
    }

    @Override
    public void add(Organiser el) {
        logger.traceEntry("saving organiser {}", el);
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("insert into Organiser (Name,Password) values (?,?)")){

            preparedStatement.setString(1, el.getName());
            preparedStatement.setString(2,el.getPassword());
            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
    }

    @Override
    public void delete(Organiser el) {
        logger.traceEntry("deleting organiser {}", el);
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("delete from Organiser where Id = ?")){

            preparedStatement.setInt(1, el.getId());
            int result = preparedStatement.executeUpdate();
            logger.trace("Removed {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
    }

    @Override
    public void update(Organiser el, Integer id) {
        logger.traceEntry("updating organiser {}", el);
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("Update Organiser set Name = ?, Password = ? where Id = ?")){

            preparedStatement.setString(1, el.getName());
            preparedStatement.setString(2, el.getPassword());
            preparedStatement.setInt(3, el.getId());
            int result = preparedStatement.executeUpdate();
            logger.trace("Updated {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
    }

    @Override
    public Organiser findById(Integer id) {
        logger.traceEntry("Finding organiser {}", id);
        Connection con = dbUtils.getConnection();
        Organiser organiser = new Organiser();
        try(PreparedStatement preparedStatement = con.prepareStatement("select * from Organiser where Id = ?")){
            preparedStatement.setInt(1, id);
            try(ResultSet result = preparedStatement.executeQuery()) {

                while(result.next()){
                    int Id = result.getInt("Id");
                    String name = result.getString("Name");
                    String password = result.getString("Password");
                    organiser.setId(Id);
                    organiser.setName(name);
                    organiser.setPassword(password);
                logger.trace("Found {} instances", organiser);
                }
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return organiser;
    }

    @Override
    public Iterable<Organiser> findAll() {
        logger.traceEntry("find all organisers");
        Connection con = dbUtils.getConnection();
        List<Organiser> organisers = new ArrayList<>();
        try(PreparedStatement preparedStatement = con.prepareStatement("select * from Organiser")){
            try(ResultSet result = preparedStatement.executeQuery()){
                while(result.next()){
                    int id = result.getInt("Id");
                    String name = result.getString("Name");
                    String password = result.getString("Password");
                    Organiser organiser = new Organiser(id,name,password);
                    organisers.add(organiser);
                }
            }
        }
        catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB" + e);
        }
        logger.traceExit(organisers);
        return organisers;
    }

    @Override
    public Collection<Organiser> getAll() {
        logger.traceEntry("get all organisers");
        Connection con = dbUtils.getConnection();
        List<Organiser> organisers = new ArrayList<>();
        try(PreparedStatement preparedStatement = con.prepareStatement("select * from Organiser")){
            try(ResultSet result = preparedStatement.executeQuery()){
                while(result.next()){
                    int id = result.getInt("Id");
                    String name = result.getString("Name");
                    String password = result.getString("Password");
                    Organiser organiser = new Organiser(id,name,password);
                    organisers.add(organiser);
                }
            }
        }
        catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB" + e);
        }
        logger.traceExit(organisers);
        return organisers;
    }

    @Override
    public Organiser findByNameAndPassword(String name, String password) {
        logger.traceEntry("Finding by name {}", name);
        Connection con = dbUtils.getConnection();
        Organiser organiser = new Organiser();
        try(PreparedStatement preparedStatement = con.prepareStatement("select * from Organiser where Name = ? and Password = ?")){
            preparedStatement.setString(1, name);
            preparedStatement.setString(2, password);
            try(ResultSet result = preparedStatement.executeQuery()) {
                while(result.next()){
                    Integer Id = result.getInt("Id");
                    String orgName = result.getString("Name");
                    String orgPass = result.getString("Password");
                    organiser.setId(Id);
                    organiser.setName(orgName);
                    organiser.setPassword(orgPass);
                    logger.trace("Found {} instances", organiser);
                }
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return organiser;
    }
}
